presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ParametricScalarImplementation.java

```diff
@@ -444,7 +447,30 @@ private void parseArguments(Method method)
                 if (implementationDependency.isPresent()) {
                     // check if only declared typeParameters and literalParameters are used
                     validateImplementationDependencyAnnotation(method, implementationDependency.get(), typeParameterNames, literalParameters);
-                    dependencies.add(createDependency(implementationDependency.get(), literalParameters));
+                    InvocationConvention invocationConvention = null;
+
+                    if (implementationDependency.get() instanceof OperatorDependency) {
```

Comment:
Move this into `createDependency`.

Make `OperatorDependency` and `FunctionDependency` use the same Convention object. To do this, you'll need to move Convention up as standalone class. This will allow you use share the code for turning it into an `InvocationConvention` object.